### PR TITLE
Mock tests http request - Closes #74

### LINKED ISSUE #74: Tests should mock http requests
See bellow the output of the "API" part of tests, esp. the last 12 tests show how long each took. Altogether the test suit takes 8 seconds when on pretty good internet connection. If I skip API tests (there is 45 of them), then the rest 228 tests take less than 2 seconds. 

There are some nice tools to do the mocking:
http://stackoverflow.com/questions/8021956/node-js-mock-http-request-and-response

```
    api
      Lisk.api()
        lisk.api()
          ✓ should create a new instance when using lisk.api()
          ✓ new lisk.api() should be Object
        #listPeers
          ✓ should give a set of the peers
        .currentPeer
          ✓ currentPeer should be set by default
        #getNethash
          ✓ Nethash should be hardcoded variables
          ✓ should give corret Nethash for testnet
        #setTestnet
          ✓ should set to testnet
        #setNode
          ✓ should be able to set my own node
          ✓ should select a node when not explicitly set
        #selectNode
          ✓ should return the node from initial settings when set
        #getRandomPeer
          ✓ should give a random peer
        #banNode
          ✓ should add current node to LSK.bannedPeers
        #getFullUrl
          ✓ should give the full url inclusive port
          ✓ should give the full url without port and with SSL
        #getURLPrefix
          ✓ should be http when ssl is false
          ✓ should be https when ssl is true
        #trimObj
          ✓ should not be equal before trim
          ✓ should be equal after trim an Object in keys and value
          ✓ should accept numbers and strings as value
        #toQueryString
          ✓ should create a http string from an object. Like { obj: "myval", key: "myval" } -> obj=myval&key=myval
        #serialiseHttpData
          ✓ should create a http string from an object and trim.
        #getAddressFromSecret
          ✓ should create correct address and publicKey
        #checkRequest
          ✓ should identify GET requests
          ✓ should identify POST requests
          ✓ should identify PUT requests
          ✓ should identify NOACTION requests
        #changeRequest
          ✓ should give the correct parameters for GET requests
          ✓ should give the correct parameters for GET requests with parameters
          ✓ should give the correct parameters for NOACTION requests
          ✓ should give the correct parameters for POST requests
          ✓ should give the correct parameters for PUT requests
        #sendRequest
          ✓ should receive Height from a random public peer (206ms)
        #listActiveDelegates
          ✓ should list active delegates (207ms)
        #listStandyDelegates
          ✓ should list non-active delegates (93ms)
        #searchDelegateByUsername
          ✓ should find a delegate by name (71ms)
        #listBlocks
          ✓ should list amount of blocks defined (81ms)
        #getBlock
          ✓ should get a block of certain height (1275ms)
        #listTransactions
          ✓ should list transactions of a defined account (70ms)
        #getTransaction
          ✓ should list a defined transaction (93ms)
        #listVotes
          ✓ should list votes of an account (144ms)
        #listVoters
          ✓ should list voters of an account (129ms)
        #getAccount
          ✓ should get account information (82ms)
        #sendLSK
          ✓ should send testnet LSK (118ms)
```

### PR DESCRIPTION
**Pull request by alepop**
Resolve #74 
- [x] Mock http request for method from issue

### CODE REVIEW

**Review by toschdev — DISMISSED** [review]
Really helpful PR, thanks!

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "mocha": "=3.2.0",
     "mocha-lcov-reporter": "=1.3.0",
     "nyc": "=10.1.2",
-    "should": "=11.1.0"
+    "should": "=11.1.0",
+    "sinon": "^2.2.0"
```
Comment (MaciejBaj) [inline package.json]: ^ -> =

File: test/api/liskApi.js [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
 	var common = require('../common');
 	var lisk = common.lisk;
+	var sinon = common.sinon;
 	process.env.NODE_ENV = 'test';
 }
 
 describe('Lisk.api()', function () {
 
 	var LSK = lisk.api();
+	var fakeResolved = function(responseData) {
```
Comment (MaciejBaj) [inline test/api/liskApi.js]: sinon provides mocked promises, please use them

File: test/api/liskApi.js [hunk 3]
```diff
@@ -454,157 +462,431 @@ describe('Lisk.api()', function () {
...
+			(callback.calledWith(expectedResponse)).should.be.true();
+			LSK.sendRequest.restore();
 		});
 	});
 
 	describe('#listStandbyDelegates', function () {
+		var expectedResponse = {
+			body: {
+				success: true,
+				delegates: [
```
Comment (MaciejBaj) [inline test/api/liskApi.js]:
```
[{
   ...
}]
```
style please

File: test/api/liskApi.js [hunk 4]
```diff
@@ -454,157 +462,431 @@ describe('Lisk.api()', function () {
...
+			LSK.sendRequest.restore();
 		});
 	});
 
 	describe('#getBlock', function () {
+		var expectedResponse = {
+			body: {
+				success: true,
+				blocks: [
+					{
```
Comment (MaciejBaj) [inline test/api/liskApi.js]: [{..}] again

File: test/api/liskApi.js [hunk 5]
```diff
@@ -454,157 +462,431 @@ describe('Lisk.api()', function () {
...
+			(callback.calledWith(expectedResponse)).should.be.true();
+			LSK.sendRequest.restore();
 		});
 	});
 
 	describe('#listBlocks', function () {
+		var expectedResponse = {
+			body: {
+				success: true,
+				blocks: [
```
Comment (MaciejBaj) [inline test/api/liskApi.js]: [{..}] again

**Review by toschdev — APPROVED** [review]
Thanks Maciej for your additional review and Aleksey for the PR

**Review by MaciejBaj — APPROVED** [review]
thanks for considering changes, good job!

**Pull request merged**